layers.go

```diff
@@ -527,7 +527,11 @@ func (r *layerStore) SetMetadata(id, metadata string) error {
 }
 
 func (r *layerStore) tspath(id string) string {
-	return filepath.Join(r.layerdir, id+tarSplitSuffix)
+	tmpPath := ""
+	if DefaultStoreOptions.GraphRoot == "" {
+		tmpPath = filepath.Join(DefaultStoreOptions.GraphRoot, "tmp")
+	}
+	return filepath.Join(tmpPath, r.layerdir, id+tarSplitSuffix)
 }
```

Comment:
I believe this is temporary data? But I could be mistaken.  The problem was that this was attempting to create the split on the readonly partition.